kernel/core/utils/stringtools.hpp

```diff
@@ -130,6 +131,241 @@ enum TimeDateFormat
+
+        // All below here is invalid
+        {".>,", NUM_INVALID}, {".>.", NUM_INVALID}, {".> ", NUM_INVALID}, {".<,", NUM_INVALID}, {".<.", NUM_INVALID}, {".< ", NUM_INVALID},
+        {",>,", NUM_INVALID}, {",>.", NUM_INVALID}, {",> ", NUM_INVALID}, {",<,", NUM_INVALID}, {",<.", NUM_INVALID}, {",< ", NUM_INVALID},
+        {" >,", NUM_INVALID}, {" >.", NUM_INVALID}, {" > ", NUM_INVALID}, {" <,", NUM_INVALID}, {" <.", NUM_INVALID}, {" < ", NUM_INVALID}
+    };
+
+    /////////////////////////////////////////////////
+    /// \brief Default Constructor of NumberFormatsVoter
+    ///
+    ///
+    /////////////////////////////////////////////////
+    NumberFormatsVoter()
+    {
+        for(size_t i = 0; i < FMT_COUNT; i++)
+            m_num_format_votes[i] = 0;
+    }
+
+    /////////////////////////////////////////////////
+    /// \brief This function is for lookup if current Tape is a valid or invalid number Format
+    ///
+    /// \param key std::string
+    /// \return int
+    ///
+    /////////////////////////////////////////////////
+    int checkNumFormat(std::string key)
+    {
+        // TODO what about encapsulation ? this should be private then
+        auto elem = num_format_lookup.find(key);
+
+        if(elem != num_format_lookup.end())
+            return elem->second;
+
+        return 0;
+    }
+
+    /////////////////////////////////////////////////
+    /// \brief This function pushes digits # inbetween seperators
+    ///
+    /// \param idx int
+    /// \return void
+    ///
+    /////////////////////////////////////////////////
+    void pushInbetween(int idx)
+    {
+        int digs_inbetween = idx - m_last_idx;
+        if(digs_inbetween > 0)
+        {
+            if(digs_inbetween < 3)
+                m_tape.push_back('<');
+            else if(digs_inbetween > 3)
+                m_tape.push_back('>');
+            else
+                m_tape.push_back('=');
+        }
+    }
+
+    /////////////////////////////////////////////////
+    /// \brief This function starts parse of a number
+    ///
+    /// \param idx int
+    /// \return void
+    ///
+    /////////////////////////////////////////////////
+    void startParseNumber(int idx)
+    {
+        m_last_idx = idx-1; //todo wie handeln wir 1ste und letzte, da diese nicht seperatoren sind
+        m_tape = "";
+        m_curr_format = 0;
+    }
+
+    /////////////////////////////////////////////////
+    /// \brief This function is called for every found seperator and adds to the tape
+    ///
+    /// \param sep char
+    /// \param idx int
+    /// \return void
+    ///
+    /////////////////////////////////////////////////
+    void addSeperator(char sep, int idx)
+    {
+        pushInbetween(idx-1);
+        m_tape.push_back(sep);
+        m_last_idx = idx;
+
+        // max 3 symbols at ones in tape
+        while(m_tape.length() > 3)
+            m_tape.erase(m_tape.begin());

```

Comment:
What about this?
```suggestion
        if (m_tape.length() > 3)
            m_tape.erase(0, m_tape.length()-3);
```
Should be faster as only one reallocation is necessary.